### DISCUSSION TITLE
POSIX ShellCheck LSP?

### DISCUSSION THREAD

**Discussion by Rudxain** [helix-editor/helix#9985] [Q&A] [answered]
> Apologies in advance if I should ask this somewhere else

There are LSPs for Bash. There's even [one that supports SC out-of-the-box](https://github.com/bash-lsp/bash-language-server). But what about other shells? Or, more specifically, POSIX-compliant ones?

~~There's a [VScode extension that provides SC](https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck) for any supported shell. Is it possible to make something similar for Helix? Would this need future plugin support?~~

~~I'm unsure if it would be appropriate to make an LSP wrapper for SC, as plugins seem like a better option~~

**Edit:**
According to this:
https://github.com/helix-editor/helix/blob/614a744d24e54225eae2ad0d27719b81c0cf9a4d/languages.toml#L877-L926
The LSP I just mentioned _should_ support almost any shell, despite the Bash-specific name.

I find it quite confusing that `name` and `scope` also seem to be Bash-specific. Why aren't their values these?
```toml
[[language]]
name = "shell"
scope = "source.sh"
```

**Comment by kirawi** [comment]
You could try plugging shellcheck into https://github.com/mattn/efm-langserver

**Reply by Rudxain** [reply]
Thanks for the suggestion. Please read the updated post. TLDR: "Bash" actually means "any shell"?

**Comment by Rudxain** [comment, marked as answer]
I forgot to mark this as answered. Both the "Bash" LSP and Shellcheck support many shells, including POSIX
